### ISSUE TITLE
Is cryptonite >= 0.23 strictly needed?

### ISSUE THREAD

**Issue by kvanbere** [tsani/servant-github-webhook#11]
When trying to build `servant-github-webhook-0.3.2.1` against `lts-7.24` you get

```
In the dependencies for servant-github-webhook-0.3.2.1:
    cryptonite-0.21 from stack configuration does not match >=0.23 (latest matching version is 0.25)
```

since `0.21` < `0.23`.

**Comment by tsani** [comment]
`lts-7.24` will also require bumping down the bounds on `memory` (probably not an issue) and `servant` / `servant-server`. It's the `servant` bound that worries me, since between version `0.8` and `0.11` a lot has happened in the servant internals, which servant-github-webhook builds off of.

EDIT: I'll lower the version bounds and try building.

**Comment by kvanbere** [comment]
I noticed there is a fork of `github-webhooks` for GHC-7.10 so my motivation is to try and eliminate the fork by supporting & building with CI the package & full example suite for ideally GHC 7.8 or at least worst case 7.10 onwards. This means I need to get `servant-github-webhook` also building with the same range since its a dependency of the servant example in my package.

Sorry for the bother, hope there's something we can do !

**Comment by rahulmutt** [comment]
@tsani @donkeybonks The changes required are minor: https://github.com/rahulmutt/servant-github-webhook/commit/f376af29032e7ec295f4488aaa58997cc85414f0

There's a use of TypeApplications that doesn't add any new power and can be replaced with a simple type annotation. The impact of removing the `forall...` parts of the instance declarations will needed to be looked at though.

I tested out my forks in a local stack project and it was working without any hitches with `lts-6.27`.

**Comment by tsani** [comment]
@donkeybonks Short answer: cryptonite-0.23 is an unnecessarily high bound.

@rahulmutt I don't think there's any impact in removing the forall in the instance declaration. I honestly don't recall why I put it there in the first place, since instance type variables are already scoped. Thank you very much for working out what changes were needed to get it to build with GHC 7.10!

I rebased @rahulmutt's ghc-7.10 branch on top of my master branch and I've ironed out the kinks in getting Travis to build against lts-10.8 and lts-6.27 (with extra-deps for things like servant and its dependencies). I lowered the bound on cryptonite to 0.19 and the bound on memory to 0.13 so that these packages won't need to be listed as extra-deps when building against lts-6.27. The tests now depend on `transformers` when building against GHC 7.10 since `Control.IO.Monad.Class` is not in base 4.9.

The only problem is that the bundled stack.yaml in the repo will have `resolver: 6.27` (and a bunch of extra-deps), so to build against other resolvers, one has to use `stack build --resolver foo`. I don't think that this will be much of an issue for deploying to stackage, since afaict stackage will perform `stack init` thus overwriting the bundled stack.yaml, but it could be an annoyance to other developers when trying to build servant-github-webhook from source.

@donkeybonks in master, I currently have a git extra-dep on @rahulmutt's fork of github-webhooks since it builds with ghc-7.10. Let me know when your upstream builds and I'll drop it.

**Comment by kvanbere** [comment]
The one on git should build, but the published 0.9.0 will not.

**Comment by tsani** [comment]
I've updated stack.yaml on master accordingly. Thanks!

**Comment by kvanbere** [comment]
I just released github-webhooks 0.9.1.

When 0.9.1 hits stack (in say, 24hrs) the explicit git dependency can be removed again.

Sorry for the inconvenience! We have mutual dependencies on each other, so this was way more awkward than I expected, ha.

**Comment by kvanbere** [comment]
Please make sure to shoot me a tag when/if you make the next release with adjusted dependency bounds so I can update the servant examples accordingly.
